File: Version3.0/SimiCartPluginFW/SimiFramework/CustomClass/SimiTable.m
//
//  SimiTable.m
//  SimiCartPluginFW
//
//  Created by Nguyen Dac Doan on 12/5/14.
//  Copyright (c) 2014 Trueplus. All rights reserved.
//

#import "SimiTable.h"

@implementation SimiTable

- (SimiSection *)addSectionWithIdentifier:(NSString *)identifier
{
    SimiSection *section = [[SimiSection alloc] initWithIdentifier:identifier];
    [self addObject:section];
    return section;
}

- (SimiSection *)addSectionWithIdentifier:(NSString *)identifier atIndex:(NSUInteger)index
{
    SimiSection *section = [[SimiSection alloc] initWithIdentifier:identifier];
    [self insertObject:section atIndex:index];
    return section;
}

- (SimiSection *)addSectionWithIdentifier:(NSString *)identifier headerTitle:(NSString *)headerTitle
{
    SimiSection *section = [[SimiSection alloc] initWithIdentifier:identifier];
    section.headerTitle = headerTitle;
    [self addObject:section];
    return section;
}

- (SimiSection *)addSectionWithIdentifier:(NSString *)identifier headerTitle:(NSString *)headerTitle footerTitle:(NSString *)footerTitle
{
    SimiSection *section = [[SimiSection alloc] initWithHeaderTitle:headerTitle footerTitle:footerTitle];
    section.identifier = identifier;
    [self addObject:section];
    return section;
}

- (SimiSection *)getSectionByIdentifier:(NSString *)identifier
{
    for (SimiSection *section in self) {
        if ([section.identifier isEqualToString:identifier]) {
            return section;
        }
    }
    return nil;
}

- (NSUInteger)getSectionIndexByIdentifier:(NSString *)identifier
{
    for (NSUInteger index = 0; index < self.count; index++) {
        SimiSection *section = [self objectAtIndex:index];
        if ([section.identifier isEqualToString:identifier]) {
            return index;
        }
    }
    return NSNotFound;
}

- (void)removeSectionsByIdentifier:(NSString *)identifier
{
    for (NSUInteger index = self.count; index > 0; ) {
        SimiSection *section = [self objectAtIndex:--index];
        if ([section.identifier isEqualToString:identifier]) {
            [self removeObjectAtIndex:index];
        }
    }
}

@end

File: simitracking/simitracking/SimiFramework/CustomClass/SimiTable.swift
//
//  SimiTable.swift
//  SimiTracking
//
//  Created by Hoang Van Trung on 1/11/17.
//  Copyright © 2017 SimiCart. All rights reserved.
//

import Foundation

class SimiTable{
    public var sections: Array = Array<SimiSection>()
    
    public func getSectionIndexById(identifier:String) -> Int{
        for section : SimiSection! in sections {
            if(section.identifier == identifier){
                return sections.index(of: section)!
            }
        }
        return 0
    }
    
    public var sectionCount: Int{
        get{
            return sections.count
        }
    }
    
    public func addSectionWithIdentifier(identifier:String) -> SimiSection{
        let section: SimiSection = SimiSection(identifier: identifier)
        self.sections.append(section)
        return section
    }
    
    public func addSection(_ section:SimiSection){
        self.sections.append(section)
    }
    
    public func sectionAtIndex(index: Int) -> SimiSection{
        return sections[index]
    }
    
}
